### PR TITLE
cpu/kinetis: add support for PWM peripherals in addition to FTM

### PR DESCRIPTION
**Pull request by benemorius**
### Contribution description
The current Kinetis PWM driver supports only FTM peripherals whereas some MCUs have PWM instead. This PR adds support for Kinetis PWM peripherals.

Most applicable boards don't have any PWM configuration in periph_conf.h and I didn't attempt to add it for boards I don't have, however I could try to address that in a followup PR.

### Testing procedure
kw41z-mini has a PWM configuration already and could be used to test this. I have only kw41z boards to test with so I haven't been able to test FTM for regressions.

### CODE REVIEW

File: cpu/kinetis/periph/pwm.c [hunk 1]
```diff
@@ -106,13 +141,40 @@ uint32_t pwm_init(pwm_t pwm, pwm_mode_t mode, uint32_t freq, uint16_t res)
                        PORT_PCR_MUX(pwm_config[pwm].chan[i].af));
         /* set the given mode */
         ftm(pwm)->CONTROLS[pwm_config[pwm].chan[i].ftm_chan].CnSC = mode;
-        /* and reset the PWM to 0% duty cycle */
-        ftm(pwm)->CONTROLS[pwm_config[pwm].chan[i].ftm_chan].CnV = 0;
```
Comment (benpicco) [inline cpu/kinetis/periph/pwm.c:110]: Why was this removed?

Comment (benemorius) [reply cpu/kinetis/periph/pwm.c:110]: Whoops. I neglected to grab the latest pwm.c from my branch before pushing this. I'll push a fixup that adds this back along with other small changes.

Incidentally, this was the beginning of my attempt to eliminate flickering during init by using a `pwm_set(); pwm_init()` paradigm. It didn't work because we can't `pwm_set()` while the hardware is declocked. I'll deal with the flicker in another PR.

Comment (benemorius) [reply cpu/kinetis/periph/pwm.c:110]: > I'll deal with the flicker in another PR.

For now I made #14721 to track this.

File: cpu/kinetis/periph/pwm.c [hunk 2]
```diff
@@ -21,29 +21,50 @@
...
-#ifdef FTM0
+/* This mode will be blocked while PWM is active */
+#ifndef PWM_PM_BLOCKER
+#define PWM_PM_BLOCKER           KINETIS_PM_STOP
+#endif
+
+/**
+ * @brief   Keeps track of whether PWM is active for power management
+ */
+static bool _is_powered_on[PWM_NUMOF];
```
Comment (benpicco) [inline cpu/kinetis/periph/pwm.c:42]: bool is a bit wasteful.
How about a bitmap

```C
static uint32_t _is_powered_on;
```

if there can be no more than 32 pwms.

File: cpu/kinetis/periph/pwm.c [hunk 3]
```diff
@@ -21,29 +21,50 @@
...
+#else /* FTM0 */
+static inline TPM_Type *tpm(pwm_t pwm)
+{
+    return pwm_config[pwm].tpm;
+}
+#endif /* FTM0 */
 
 static void poweron(pwm_t pwm)
 {
+    if (_is_powered_on[pwm]) {
```
Comment (benpicco) [inline cpu/kinetis/periph/pwm.c:60]:
```suggestion
    if (_is_powered_on & (1 << pwm)) {
```

File: cpu/kinetis/periph/pwm.c [hunk 4]
```diff
@@ -21,29 +21,50 @@
...
+        return;
+    }
+
+    pm_block(PWM_PM_BLOCKER);
+    _is_powered_on[pwm] = true;
```
Comment (benpicco) [inline cpu/kinetis/periph/pwm.c:65]:
```suggestion
   bit_set32(&_is_powered_on, pwm);
```

**Comment by benpicco** [comment]
Sorry this has fallen through the cracks - please squash!

**Comment by benemorius** [comment]
Very sorry I've fallen through the cracks myself lately! I anticipate I'll resume normal activities within the next few months.
